Pull request for updated Readme.MD with business context (#8)
alliander-opensource/der-scheduling: README.md

## Targeted applicability
- DER = Distributed Energy Resources (e.g. PV solar, Windpower, Battery infeed into distribution grid)
- EMS = (Plant / Industrial) Energy Managment Systems
- HEMS = Home Energy Management Systems
- Specific solutions such as the [FNN-steurbox](https://www.vde.com/de/fnn/arbeitsgebiete/digitalisierung-metering/lastenhefte/steuerbox) in Germany

## Why IEC scheduling?
IEC scheduling solves two structural problems in controling complex energy systems where more than one actor and more than one use-case is involved (*...and that is nearly almost the case*).

**Problem 1**: There are many controls from remote, and in the local device itself. How do I get myself into control what is happening with all those loose controls? Otherwise, I won't understand the behavior and lose control.

![Several actors with several capacity envelopes](./images/readme_totalenvelopes.png)

> **Answer**: IEC scheduling completely changes the concept of controlling, compared to what is known till sofar in the IEC landscape for energy automation. The main principle of it is to place ALL controls into parallell schedules and where a policy defines the priority of those schedules. There are NO direct controls. All controls, even locally generated are placed in (separate) schedules. Included in defining this policy is to give priority to schedules dependent on situational circumstances, such as:

Reviewer: ```suggestion
**Answer**: IEC scheduling completely changes the concept of controlling, compared to what is known till sofar in the IEC landscape for energy automation. The main principle of it is to place ALL controls into parallell schedules. The priority of those schedules is defined through policies. There are NO direct controls. All controls, even locally generated are placed in (separate) schedules. Included in defining this policy is to give priority to schedules dependent on situational circumstances, such as:
```

> **Answer**: IEC scheduling completely changes the concept of controlling, compared to what is known till sofar in the IEC landscape for energy automation. The main principle of it is to place ALL controls into parallell schedules and where a policy defines the priority of those schedules. There are NO direct controls. All controls, even locally generated are placed in (separate) schedules. Included in defining this policy is to give priority to schedules dependent on situational circumstances, such as:

Reviewer: The last sentence of the answer starting with `Included in defining this policy ...` is not clear to me. I think the word including leads to the interpretation: policies are just another schedule.

1. Device just starting up, so soft-start may get priorities
2. (Internet)communication is down, so slowly move towards pre-defined (safe) schedules.
3. Voltage is too high, so curtailment of infeed get priority (and parallel simulate with HEMS to use more energy locally)

![Several actors with several IEC-schedules](./images/readme_totalschedules.png)

**Problem 2**:  What communication protocol to use, because my use-case needs "protocol y", my DSO needs "protocol z" and for car-charging I need another one. Please don't create or force me to use another communication protocol, do you?

> **Answer**: Although the standard for Scheduling is under the IEC 61850 standard, however the funcion scheduling itself isn't requiering the IEC 61850 communication protocol use.  It doesn't matter as long as you can manage a schedule to the scheduling logic, whatever protocol you use. This will enable you to choose your protocol which is needed for your use-case. And yes, other requirements to realize the right security level should be applied to all protocols.

Reviewer: ```suggestion
**Answer**: Although the standard for Scheduling is under the IEC 61850 standard, the funcion scheduling itself isn't requiering the IEC 61850 communication protocol use.  It doesn't matter as long as you can manage a schedule to the scheduling logic, whatever protocol you use. This will enable you to choose your protocol which is needed for your use-case. And yes, other requirements to realize the right security level should be applied to all protocols.
```

![Infographic embedding IEC-scheduling into Fledge](./images/readme_infographic_embedding.png)

## Roadmap
### Scheduling as stand alone-service (realized!)
This has been realized till sofar. The main functionalities are to control a DERs regarding:
- absolute power output
- maximal power output
- turning them on or off

> The configuration of how the software is configured is realized with the [SCL file](https://github.com/alliander-opensource/der-scheduling/blob/main/models/der_scheduler.cid) which can be managed and adopted by our needs by [CoMPAS](https://github.com/com-pas). Schedules themselves are now fed into the software with an API from the CLI.

Reviewer 2: You could mention that using CoMPAS is just an option and any IEC 61850 compliant SCL tool could be used for the configuration.

Author: Changed it. Thanks for suggestion

With positive results, the solution is benchmarked against German commercial FNN-steuerboxes to have the same behaviour and to have at least the same performance.

### Scheduling integraded with official communication protol (active!)

> We currently integrate the IEC scheduling as stand-service into [FLEDGE](https://www.lfedge.org/projects/fledge/) in close collaboration with [FLEDGE POWER](https://lfenergy.org/projects/fledgepower/) as they share the vision to build productive software for the edge. (Please remark that "scheduling" in Fledge is not the same as IEC-scheduling or DER-scheduling developed here). In this stage we expect to communicate with one protocol (WAN) to central systems (SCADA / Substation automation like) and with one protocol (LAN and probably MQTT or Modbus) to local devices.

Reviewer 2: > We currently integrate the IEC scheduling as stand-service
Should be stand-alone-service?

Reviewer 2: > ### Scheduling integraded with official communication protol (active!)
### Scheduling integrated with official communication protocol (active!)
